packages/react-components/src/components/table-output-component.tsx

```diff
@@ -653,4 +655,126 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
+                const startIndex = await this.fetchTableIndex(this.startTimestamp < this.endTimestamp ? this.startTimestamp : this.endTimestamp);
+                const endIndex = await this.fetchTableIndex(this.startTimestamp > this.endTimestamp ? this.startTimestamp : this.endTimestamp);
+
+                if (startIndex && endIndex) {
+                    fetchIndex = startIndex;
+                    totalLinesToFetch = endIndex - startIndex;
+                }
+            }
+        }
+
+        let fetchLinesRemainder = totalLinesToFetch;
+
+        while (fetchLinesRemainder > 0) {
+            let curLinesToFetch = bufferSize;
+
+            if (fetchLinesRemainder > bufferSize) {
+                fetchLinesRemainder-=bufferSize;
+            } else {
+                curLinesToFetch = fetchLinesRemainder;
+                fetchLinesRemainder = 0;
+            }
+
+            const tspClientResponse = await tspClient.fetchTableLines(traceUUID, outputId, QueryHelper.tableQuery(this.columnIds, fetchIndex, curLinesToFetch,
+                { [QueryHelper.REQUESTED_TABLE_COLUMN_IDS_KEY]: columnsIds }));
+            const lineResponse = tspClientResponse.getModel();
+            if (!tspClientResponse.isOk() || !lineResponse) {
+                return;
+            }
+            const model = lineResponse.model;
+            const lines = model.lines;
+            lines.forEach(line => {
+                const obj: any = {};
+                const cells = line.cells;
+                const columnIds = model.columnIds;
+
+                if (this.showIndexColumn) {
+                    obj[0] = line.index.toString();
+                }
+
+                cells.forEach((cell, index) => {
+                    const id = this.showIndexColumn ? columnIds[index] + 1 : columnIds[index];
+                    obj[id] = cell.content;
+                });
+
+                linesArray.push(obj);
+            });
+
+            for (let i=0;i<lines.length;i++) {
+                // Stores each csv row data
+                const csvrow = [];
+                for (let j=0;j<lines[i].cells.length;j++) {
+                    // Get the text data of each cell of
+                    // a row and push it to csvrow
+                    const cellContent = lines[i].cells[j].content;
+                    if (cellContent.includes(',')) {
+                        csvrow.push('"' + cellContent + '"');
+                    } else {
+                        csvrow.push(cellContent);
+                    }
+                }
+                // Combine each column value with comma
+                csv_data.push(csvrow.join(','));
+            }
+        }
+
+        // combine each row data with new line character
+        const tableString = csv_data.join('\n');
+
+        const link = document.createElement('a');
+        link.setAttribute('href', `data:text/csv;charset=utf-8,${encodeURIComponent(tableString)}`);
+        link.setAttribute('download', 'export');
```

Comment:
I would have expected a file chooser dialog for the user to decide where to store it and what to call it. Maybe for future improvement. But otherwise perhaps the name should be computed from the trace name? Now it's export (1), export (2), etc...